pkg/test/framework/components/istio/ingress.go

```diff
@@ -182,103 +182,26 @@ func (c *ingressImpl) callEcho(options echo.CallOptions, retry bool, retryOption
 			return nil, fmt.Errorf("protocol %v not supported, provide explicit port", options.Port.Protocol)
 		}
 	}
-	host := options.Host
-	// Default host based on protocol
-	switch options.Port.Protocol {
-	case protocol.HTTP:
-		options.Host = c.HTTPAddress().IP.String()
-	case protocol.HTTPS:
-		options.Host = c.HTTPSAddress().IP.String()
-	case protocol.TCP:
-		options.Host = c.TCPAddress().IP.String()
-	default:
-		return nil, fmt.Errorf("protocol %v not supported, provide explicit port", options.Port.Protocol)
+	if len(options.Host) == 0 {
+		// Default host based on protocol
+		switch options.Port.Protocol {
+		case protocol.HTTP:
+			options.Host = c.HTTPAddress().IP.String()
+		case protocol.HTTPS:
+			options.Host = c.HTTPSAddress().IP.String()
+		case protocol.TCP:
+			options.Host = c.TCPAddress().IP.String()
+		default:
+			return nil, fmt.Errorf("protocol %v not supported, provide explicit port", options.Port.Protocol)
+		}
 	}
 	if options.Headers == nil {
 		options.Headers = map[string][]string{}
 	}
-	options.Headers["Host"] = []string{host}
-	return common.CallEcho(&options, retry, retryOptions...)
-}
-
-func (c *ingressImpl) Call(options ingress.CallOptions) (ingress.CallResponse, error) {
-	return c.call(options, false)
-}
-
-func (c *ingressImpl) call(options ingress.CallOptions,
-	retry bool, retryOptions ...retry.Option) (ingress.CallResponse, error) {
-	if err := options.Sanitize(); err != nil {
-		scopes.Framework.Fatalf("CallOptions sanitization failure, error %v", err)
-	}
-	req := &echo.CallOptions{
-		Port: &echo.Port{
-			Protocol:    protocol.HTTP,
-			ServicePort: options.Address.Port,
-		},
-		Host: options.Address.IP.String(),
-		Path: options.Path,
-		Headers: map[string][]string{
-			"Host": {options.Host},
-		},
-	}
-
-	for k, v := range options.Headers {
-		req.Headers[k] = v
+	if _, hasHost := options.Headers["Host"]; !hasHost {
+		options.Headers["Host"] = []string{options.Host}
```

Comment:
I am a bit confused by this change. I think we want to set the Host header to something like `foo.bar` not `1.2.3.4` to test the gateway routing?

Reply:
Ah I see what you did - but i am not sure why, this seems like a degraded UX. Was there some reason the old logic doesn't work now?

Reply:
The API was confusing ... I cleaned it up. I renamed `Host` to `Address` to clarify. PTAL